pkg/parser/parser.go

```diff
@@ -247,8 +247,8 @@ func (p *Parser) parseAssignmentStatement(scope *scope) Node {
 
 func (p *Parser) parseAssignable(scope *scope) Node {
 	name := p.cur.Literal
-	p.advance()
 	v, ok := scope.get(name)
+	p.advance()
```

Comment:
did this line move for a reason? it splits up the assignment of `ok` and the test of it.

Reply:
yeah, the thought was that the error message is linked to the position after the variable, rather on it. but i didn't fix it properly, i'll reworked it again in a separate commit.

Reply:
ok, done.